pkg/handler/csv/loki_csv.go

```diff
@@ -0,0 +1,98 @@
+package csv
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/netobserv/network-observability-console-plugin/pkg/model"
+	"github.com/netobserv/network-observability-console-plugin/pkg/utils"
+)
+
+const (
+	timestampCol = "Timestamp"
+)
+
+func GetCSVData(qr *model.QueryResponse, columns []string) ([][]string, error) {
+	if columns != nil && len(columns) == 0 {
+		return nil, fmt.Errorf("columns can't be empty if specified")
+	}
+
+	if streams, ok := qr.Data.Result.(model.Streams); ok {
+		return manageStreams(streams, columns)
+	}
+	return nil, fmt.Errorf("loki returned an unexpected type: %T", qr.Data.Result)
+}
+
+func manageStreams(streams model.Streams, columns []string) ([][]string, error) {
+	//make csv datas containing header as first line + rows
+	datas := make([][]string, 1)
+	//set Timestamp as first data
+	if columns == nil || utils.Contains(columns, timestampCol) {
```

Comment:
since `utils.Contains` uses a `for` and Go allows iterating a `nil` slice, the `columns == nil ||` is not needed.

On the other side, I see that this function is invoked multiple times inside the loop and might be slow because looking up a value in an array is `O(n)`. I think that we should first move the `columns []string` contents to a `map[string]struct{}` for an `O(1)` lookup.

Reply:
+1, I see that there's a function `utils.GetMapInterface` that can be reused for array to map conversion. By the way, any reason why to use `map[string]struct{}` rather than `map[string]interface{}` ? (hence filling with `struct{}` rather than `nil`)